uast/utils.go

```diff
@@ -17,3 +17,34 @@ func Tokens(n *Node) []string {
 	}
 	return tokens
 }
+
+// CyclomaticComplexity returns the cyclomatic complexity for the node. This uses the method of
+// counting one + one of the following UAST Roles if present on any children:
+// If | SwitchCase |  SwitchDefault | For[Each] | [Do]While | TryCatch | Continue | OpBoolean* | Goto
+// Important: since some languages allow for code defined
+// outside function definitions, this won't check that the Node has the role FunctionDeclarationRole
+// so the user should check that if the intended use is calculating the complexity of a function/method.
+// If the children contain more than one function definitions, the value will not be averaged between
+// the total number of function declarations but given as a total.
+func CyclomaticComplexity(n *Node)  int {
+	complx := 1
+
+	iter := NewOrderPathIter(NewPath(n))
+
+	for {
+		p := iter.Next()
+		if p.IsEmpty() {
+			break
+		}
+		n := p.Node()
+		for _, r := range n.Roles {
+			switch(r) {
+			case If, SwitchCase, SwitchDefault, For, ForEach, While,
```

Comment:
There are several definitions, I went with the most common. IfElse/Break/Continue and exceptions are almost never considered (I was surprised about IfElse too) , Return never and the switch "default" clause is probably 50%. Goto is definitely needed. The boolean operators are definitely not binary in lots of languages as you say, even in some that use infix (like Python) use lists in the AST, I think I've a PR about that. What is also important is to know if the boolean evaluation on tests is done on short circuit because that definitely  increases the complexity (one path for operator vs one path for the test). I accounted them as short circuit since its the most common option but we should update once we have some way to represent these language invariants in the AST or metadata.